server/channels/store/sqlstore/channel_store.go

```diff
@@ -2021,6 +2021,92 @@ func (s SqlChannelStore) GetChannelMembersTimezones(channelId string) ([]model.S
 	return dbMembersTimezone, nil
 }
 
+func (s SqlChannelStore) GetChannelsWithUnreadsAndWithMentions(ctx context.Context, channelIDs []string, userID string) ([]string, []string, map[string]int64, error) {
+	query := s.getQueryBuilder().Select(
```

Comment:
Not sure how performant is this query, but at least it is only one query, way better than before :P

Reply:
This is quite performant. ChannelMembers has a PK on channelID+userID so should be fine. In case you have doubts on how performant a query is, always do an `EXPLAIN ANALYZE` and check.